DTML.EduBot/Dialogs/RootDialog.cs

```diff
@@ -3,76 +3,265 @@
+            }
+        }
+
+        private async Task AfterDialogEnded(IDialogContext context, IAwaitable<object> result)
+        {
+            await Conversation.SendAsync(context.MakeMessage(), () => this);
+        }
+
+        private async Task AfterChoosingLanguageSwitch(IDialogContext context, IAwaitable<object> result)
+        {
+            var response = await result as string;
+
+            try
+            {
+                using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+                {
+                    UserData.UserData userData = scope.Resolve<IUserDataRepository>().GetUserData(context.Activity.From.Id);
+
+                    string translatedYes = await MessageTranslator.TranslateTextAsync(Yes, userData.NativeLanguageIsoCode);
+                    string translatedNo = await MessageTranslator.TranslateTextAsync(No, userData.NativeLanguageIsoCode);
+
+                    if (translatedYes.Equals(response, StringComparison.OrdinalIgnoreCase))
+                    {
+                        string translatedSelfIntroduction =
+                            await MessageTranslator.TranslateTextAsync(BotPersonality.BotSelfIntroduction,
+                                userData.NativeLanguageIsoCode);
+                        await context.PostAsync($"{translatedSelfIntroduction}");
+
+                        string translatedUserNameQuestion =
+                            await MessageTranslator.TranslateTextAsync(BotPersonality.UserNameQuestion,
+                                userData.NativeLanguageIsoCode);
+                        await context.PostAsync($"{translatedUserNameQuestion}");
+
+                        context.Wait(this.UserNameReceivedInNativeLanguageAsync);
+                    }
+                    else if (translatedNo.Equals(response, StringComparison.OrdinalIgnoreCase))
+                    {
+                        await context.PostAsync($"{BotPersonality.UserNameQuestion}");
+
+                        context.Wait(this.UserNameReceivedAsync);
+                    }
+                }
+            }
+            catch (TooManyAttemptsException)
+            {
+                using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+                {
+                    UserData.UserData userData =
+                        scope.Resolve<IUserDataRepository>().GetUserData(context.Activity.From.Id);
+                    
+                    string translatedTooManyAttemptMessage = await MessageTranslator.TranslateTextAsync(TooManyAttemptMessage, userData.NativeLanguageIsoCode);
+
+                    await context.PostAsync($"{translatedTooManyAttemptMessage}");
+                }
+                await Conversation.SendAsync(context.MakeMessage(), () => this);
+            }
+        }
+
+        private async Task<ICollection<string>> TranslatedChoices(IEnumerable<string> choices, string languageToTranslate)
```

Comment:
This should probably be part of a different class, maybe the Translator class? We will want all dialogs to be able to have access to this functionality